AppSearchSample/app/src/main/java/com/android/example/appsearchsample/NoteListItemAdapter.kt

```diff
@@ -84,7 +86,9 @@ class NoteListItemAdapter(private val onDelete: (SearchResult?) -> Unit) :
       }
 
       override fun areContentsTheSame(oldItem: SearchResult, newItem: SearchResult): Boolean {
-        return oldItem == newItem
+        return oldItem.genericDocument == newItem.genericDocument &&
+          oldItem.genericDocument.toDocumentClass(Note::class.java) ==
```

Comment:
I believe checking for GenericDocument equality should be sufficient -- converting it to a document class would just compare a subset of fields of the GenericDocument.